src/main/java/executor/service/service/impl/DIFactory.java

```diff
@@ -32,48 +32,106 @@
+            object = new ProxyValidatorImpl(propertiesConfig);
+        } else if (clazz == ProxySourceClient.class) {
+            ProxyProvider proxyProvider = createObject(ProxyProvider.class);
+            ProxyValidator proxyValidator = createObject(ProxyValidator.class);
+            PropertiesConfig propertiesConfig = createObject(PropertiesConfig.class);
+            object = new ProxySourceClientImpl(proxyProvider, proxyValidator, propertiesConfig);
+        } else if (clazz == ScenarioSourceQueue.class) {
+            object = new ScenarioSourceQueue();
+        } else if (clazz == ScenarioTaskWorker.class) {
+            ScenarioSourceListener scenarioSourceListener = createObject(ScenarioSourceListener.class);
+            ScenarioSourceQueue scenarioSourceQueue = createObject(ScenarioSourceQueue.class);
+            object = new ScenarioTaskWorker(scenarioSourceListener, scenarioSourceQueue);
+        } else if (clazz == ProxySourceQueue.class) {
+            object = new ProxySourceQueue();
+        } else if (clazz == ProxyTaskWorker.class) {
+            ProxySourceClient proxySourceClient = createObject(ProxySourceClient.class);
+            ProxySourceQueue proxySourceQueue = createObject(ProxySourceQueue.class);
+            object = new ProxyTaskWorker(proxySourceClient, proxySourceQueue);
+        } else if (clazz == WebDriverInitializer.class) {
+            WebDriverConfig webDriverConfig = createObject(WebDriverConfig.class);
+            object = new WebDriverInitializerImpl(webDriverConfig);
+        } else if (clazz == ExecutorService.class) {
+            PropertiesConfig propertiesConfig = createObject(PropertiesConfig.class);
+            object = new ThreadPoolExecutorObject(propertiesConfig).threadPoolExecutor();
+        } else if (clazz == ExecutionService.class) {
+            ScenarioExecutor scenarioExecutor = createObject(ScenarioExecutor.class);
+            WebDriverInitializer webDriverInitializer = createObject(WebDriverInitializer.class);
+            object = new ExecutionServiceImpl(scenarioExecutor, webDriverInitializer);
+        } else if (clazz == TasksFactory.class) {
+            ScenarioTaskWorker scenarioTaskWorker = createObject(ScenarioTaskWorker.class);
+            ProxyTaskWorker proxyTaskWorker = createObject(ProxyTaskWorker.class);
+            object = new TasksFactoryImpl(scenarioTaskWorker, proxyTaskWorker);
+        } else if (clazz == ParallelFlowExecutorService.class) {
+            ExecutorService threadPoolExecutor = createObject(ExecutorService.class);
+            ExecutionService executionService = createObject(ExecutionService.class);
+            TasksFactory tasksFactory = createObject(TasksFactory.class);
+            object = new ParallelFlowExecutorServiceImpl(threadPoolExecutor, executionService, tasksFactory);
         }
+
+        if (object != null) {
+            cache.put(clazz, object);
+            return (T) object;
+        }
+
         return null;
```

Comment:
Returning null can be confusing, since we don’t know at what point the service will be needed, and what exactly will be the cause of the error, and at first you will looking why is null and where he placed. Do you think it would be better to throw the error?

Reply:
I like this option better:

if (object != null) {
            cache.put(clazz, object);
            return (T) object;
        } else {
            throw new RuntimeException("Failed to create an instance of " + clazz.getName());
        }